tests/upgrades/test_client.py

```diff
@@ -22,92 +22,51 @@
 import time
 
 import pytest
-from fabric.api import execute
-from upgrade.helpers.docker import docker_execute_command
-from upgrade_tests.helpers.scenarios import dockerize
+from broker import Broker
 
 from robottelo.config import settings
-from robottelo.constants import DEFAULT_CV
-from robottelo.constants import DEFAULT_SUBSCRIPTION_NAME
 from robottelo.constants import FAKE_0_CUSTOM_PACKAGE
 from robottelo.constants import FAKE_4_CUSTOM_PACKAGE
-from robottelo.constants import REPOS
-
-
-# host machine for containers
-docker_vm = settings.upgrade.docker_vm
-
-
-@pytest.fixture(scope='module')
-def pre_upgrade_repo(module_product, module_target_sat):
-    """Enable custom errata repository"""
-    pre_upgrade_repo = module_target_sat.api.Repository(
-        url=settings.repos.yum_0.url, product=module_product
-    ).create()
-    assert pre_upgrade_repo.sync()['result'] == 'success'
-    return pre_upgrade_repo
+from robottelo.hosts import ContentHost
+from robottelo.utils.installer import InstallerCommand
 
 
 @pytest.fixture(scope='module')
-def post_upgrade_repo(module_product, module_target_sat):
-    """Enable custom errata repository"""
-    post_upgrade_repo = module_target_sat.api.Repository(
-        url=settings.repos.yum_9.url, product=module_product
-    ).create()
-    assert post_upgrade_repo.sync()['result'] == 'success'
-    return post_upgrade_repo
+def sat_with_katello_agent(module_target_sat):
+    """Enable katello-agent on module_target_sat"""
+    module_target_sat.register_to_cdn()
+    # Enable katello-agent from satellite-installer
+    result = module_target_sat.install(
+        InstallerCommand('foreman-proxy-content-enable-katello-agent true')
+    )
+    assert result.status == 0
+    # Verify katello-agent is enabled
+    result = module_target_sat.install(
+        InstallerCommand(help='| grep foreman-proxy-content-enable-katello-agent')
+    )
+    assert 'true' in result.stdout
+    yield module_target_sat
 
 
-@pytest.fixture(scope='module')
-def pre_upgrade_cv(default_org, pre_upgrade_repo, module_target_sat):
-    """Create and publish repo"""
-    pre_upgrade_cv = module_target_sat.api.ContentView(
-        organization=default_org, repository=[pre_upgrade_repo.id]
-    ).create()
-    pre_upgrade_cv.publish()
-    pre_upgrade_cv = pre_upgrade_cv.read()
-    return pre_upgrade_cv
+@pytest.fixture(scope="module")
+def rhel_content_host(request):
+    """A function-level fixture that provides a content host object parametrized"""
+    return Broker(container_host='ubi8:latest', host_class=ContentHost).checkout()
```

Comment:
Why don't we use a content host from `pytest_fixtures/core/contenthosts.py`?

Reply:
We do not want to check-in the container as it will use post checks after upgrade

Reply:
Makes sense. Could you add that to the docstring, so it is more obvious why we aren't use the normal content hosts. You could even consider moving this under the file I mentioned above so there's more visibility for other people doing similar testing.